src/Symfony/Component/Security/Core/Test/AccessDecisionStrategyTestCase.php

```diff
@@ -75,6 +94,37 @@ public function vote(TokenInterface $token, $subject, array $attributes): int
             {
                 return $this->vote;
             }
+
+            public function __call($function, $args)
+            {
+                throw new LogicException('This function must not be acceded.');
+            }
+        };
+    }
+
+    final protected static function getVoterWithVoteObject(int $vote): VoterInterface
+    {
+        return new class($vote) implements VoterInterface {
+            public function __construct(
+                private int $vote,
+            ) {
+            }
+
+            public function vote(TokenInterface $token, $subject, array $attributes): int
+            {
+                return $this->vote;
+            }
+
+            public function getVote(TokenInterface $token, mixed $subject, array $attributes): Vote
+            {
+                return new Vote($this->vote);
+            }
         };
     }
+
+    final protected static function getAccessDecision(bool $decision, array $votes): AccessDecision
+    {
+        return new AccessDecision($decision ? VoterInterface::ACCESS_GRANTED : VoterInterface::ACCESS_DENIED,
+            array_map(fn ($v) => new Vote($v), $votes));
```

Comment:
it could be convenient to allow Vote objects in the array also (+ add some phpdoc)